Unit Tests/Tests/Redux/Reducers/User Subscription/AddedToRoomReducerTests.swift

```diff
@@ -0,0 +1,176 @@
+                    isPrivate: false,
+                    pushNotificationTitle: nil,
+                    customData: nil,
+                    lastMessageAt: .distantPast,
+                    readSummary: ReadSummaryState(
+                        unreadCount: 30
+                    ),
+                    createdAt: .distantPast,
+                    updatedAt: .distantPast
+                )
+            ]
+        )
+        
+        let stubReducer = StubReducer<Reducer.Model.RoomList>(reduce_stateToReturn: reducer_stateToReturn, reduce_expectedCallCount: 1)
+        
+        let dependencies = DependenciesDoubles(roomListReducer: stubReducer.reduce)
+        
+        /******************/
+        /*----- WHEN -----*/
+        /******************/
+        
+        let outputState = Reducer.UserSubscription.AddedToRoom.reduce(action: action, state: inputState, dependencies: dependencies)
+        
+        /******************/
+        /*----- THEN -----*/
+        /******************/
+        
+        let expectedState = MasterState(
+            currentUser: TestState.user,
+            joinedRooms: RoomListState(
+                rooms: [
+                    "first-room" : RoomState(
+                        identifier: "first-room",
+                        name: "First",
+                        isPrivate: false,
+                        pushNotificationTitle: nil,
+                        customData: nil,
+                        lastMessageAt: .distantPast,
+                        readSummary: .empty,
+                        createdAt: .distantPast,
+                        updatedAt: .distantPast
+                    ),
+                    "second-room" : RoomState(
+                        identifier: "second-room",
+                        name: "Second",
+                        isPrivate: false,
+                        pushNotificationTitle: nil,
+                        customData: nil,
+                        lastMessageAt: .distantPast,
+                        readSummary: .empty,
+                        createdAt: .distantPast,
+                        updatedAt: .distantPast
+                    ),
+                    "third-room" : RoomState(
+                        identifier: "third-room",
+                        name: "Third",
+                        isPrivate: false,
+                        pushNotificationTitle: nil,
+                        customData: nil,
+                        lastMessageAt: .distantPast,
+                        readSummary: ReadSummaryState(
+                            unreadCount: 30
+                        ),
+                        createdAt: .distantPast,
+                        updatedAt: .distantPast
+                    )
+                ]
+            ),
+            users: TestState.userList
+        )
+        
+        XCTAssertEqual(outputState, expectedState)
+        XCTAssertEqual(stubReducer.reduce_actualCallCount, 1)
```

Comment:
**Some general points on all your reducer tests:**

- keep your inputs and outputs as minimal as possible. I've noticed some of the tests are unnecessarily verbose.  Example below. 

- When defining `expectedState`, rather then rewriting out raw values, if you're expecting a certain State property to be...
   - unchanged, then use `inputState.property` for the value
   - changed, then you can probably use `reduce_stateToReturn` for the value

- the only thing that needs ascerting in the `UserSubscription` reducer tests is that the sub-reducers are invoked with the appropriate arguments.  So 

   - Its not essential to emulate what would happen in the real world with your input and output data.  Just ensure that things that might change do actually change in the output
   - Don't forget to make these assertions:

```
	XCTAssertEqual(stubReducer.reduce_stateLastReceived, inputState.joinedRooms)
	XCTAssertTrue(stubReducer.reduce_actionLastReceived === action)  // <-- this is ugly because of `class Action`
```

--------------------------

So to take this specific `test_reduce_withCurrentStateAndAddedToRoomAction_returnsStateFromDedicatedReducer` test as an example.  

I have been able to improve the assertions and reduce the lines of code in this single test from 160+ to 75 as follows:

```
        /******************/
        /*---- GIVEN -----*/
        /******************/
        
        let inputState = MasterState(
            currentUser: .empty,
            joinedRooms: .empty,
            users: .empty
        )
        
        let action = AddedToRoomAction(
            event: Wire.Event.AddedToRoom(
                room: Wire.Room(
                    identifier: "new-room",
                    name: "New",
                    createdById: "random-user",
                    isPrivate: false,
                    pushNotificationTitleOverride: nil,
                    customData: nil,
                    lastMessageAt: .distantPast,
                    createdAt: .distantPast,
                    updatedAt: .distantPast,
                    deletedAt: .distantPast),
                membership: Wire.Membership(
                    roomIdentifier: "new-room",
                    userIdentifiers: []
                ),
                readState: Wire.ReadState(
                    roomIdentifier: "new-room",
                    unreadCount: 30,
                    cursor: nil
                )
            )
        )
        
        let reducer_stateToReturn = RoomListState(
            rooms: [
                "new-room" : RoomState(
                    identifier: "new-room",
                    name: "New",
                    isPrivate: false,
                    pushNotificationTitle: nil,
                    customData: nil,
                    lastMessageAt: .distantPast,
                    readSummary: .empty,
                    createdAt: .distantPast,
                    updatedAt: .distantPast
                )
            ]
        )
        
        let stubReducer = StubReducer<Reducer.Model.RoomList>(reduce_stateToReturn: reducer_stateToReturn, reduce_expectedCallCount: 1)
        
        let dependencies = DependenciesDoubles(roomListReducer: stubReducer.reduce)
        
        /******************/
        /*----- WHEN -----*/
        /******************/
        
        let outputState = Reducer.UserSubscription.AddedToRoom.reduce(action: action, state: inputState, dependencies: dependencies)
        
        /******************/
        /*----- THEN -----*/
        /******************/
        
        let expectedState = MasterState(
            currentUser: inputState.currentUser,
            joinedRooms: reducer_stateToReturn,
            users: inputState.users
        )
        
        XCTAssertEqual(outputState, expectedState)
        XCTAssertEqual(stubReducer.reduce_actualCallCount, 1)
        XCTAssertEqual(stubReducer.reduce_stateLastReceived, inputState.joinedRooms)
        XCTAssertTrue(stubReducer.reduce_actionLastReceived === action)
```